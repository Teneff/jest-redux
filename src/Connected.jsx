import React from "react";
import { connect } from "react-redux";
import {
  SOMETHING_SIGNIFICANT_HAPPENED,
  SOMETHING_INSIGNIFICANT_HAPPENED
} from "./redux/actionTypes";

const Connected = ({ onSignificant, onInsignificant, dates }) => {
  return (
    <div className="connected">
      <h2>{"<Connected />"}</h2>
      <div className="columns">
        {Object.entries(dates)
          .filter(([k, v]) => Array.isArray(v))
          .map(([key, values]) => (
            <section
              key={key}
              onClick={() => {
                key === "significant" ? onSignificant() : onInsignificant();
              }}
            >
              <h3>{key}</h3>
              <ul>
                {values.map(val => (
                  <li>{val.toString()}</li>
                ))}
              </ul>
            </section>
          ))}
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  ...state.something
});

const mapDispatchToProps = dispatch => ({
  onSignificant: () => {
    dispatch({
      type: SOMETHING_SIGNIFICANT_HAPPENED
    });
  },
  onInsignificant: () => {
    dispatch({
      type: SOMETHING_INSIGNIFICANT_HAPPENED
    });
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Connected);
