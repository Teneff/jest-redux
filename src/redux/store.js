import { combineReducers, createStore } from "redux";
import something from "./reducer/something";
import { devToolsEnhancer } from "redux-devtools-extension";

export default createStore(
  combineReducers(
    {
      something
    },
    process.env.NODE_ENV === 'development' ? devToolsEnhancer() : undefined
  )
);
