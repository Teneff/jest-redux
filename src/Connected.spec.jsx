import React from "react";
import Connected from "./Connected";
import store from "./redux/store";
import { shallow } from "enzyme";
import { Provider, ReactReduxContext } from "react-redux";

describe("<Connected />", () => {
  describe("wrapped in <Provider />", () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <Provider store={store}>
          <Connected />
        </Provider>
      ).dive();
    });

    it("should match the snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("passing options wrappingComponent", () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<Connected />, {
        wrappingComponent: Provider,
        wrappingComponentProps: {
          store,
          context: ReactReduxContext
        }
      });
    });

    it("should match the snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("passing context store", () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<Connected />, {
        context: store,
      });
    });

    it("should match the snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
