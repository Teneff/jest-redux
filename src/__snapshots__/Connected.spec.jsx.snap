// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Connected /> passing context store should match the snapshot 1`] = `undefined`;

exports[`<Connected /> passing options wrappingComponent should match the snapshot 1`] = `undefined`;

exports[`<Connected /> wrapped in <Provider /> should match the snapshot 1`] = `ShallowWrapper {}`;
